File: Inheritance.py
class BangunDatar():
    def luas(self):
        print("Menghitung Luas bangun datar")

    def keliling(self):
        print("Menghitung Keliling bangun datar")


class Persegi(BangunDatar):
    def _init_(self, sisi=0):
        self.sisi = sisi

    def luas(self):
        print("Luas Persegi: ", self.sisi * self.sisi)

    def keliling(self):
        print("Keliling Persegi: ", 4 * self.sisi)


class Lingkaran(BangunDatar):

    def _init_(self, r=0):
        self.r = r

    def luas(self):
        print("Luas Lingkaran: ", 3.14 * self.r ** 2)

    def keliling(self):
        print("Keliling Lingkaran: ", 2 * 3.14 * self.r)


class PersegiPanjang(BangunDatar):
    def _init_(self, panjang=0, lebar=0):
        self.panjang = panjang
        self.lebar = lebar

    def luas(self):
        print("Luas Persegi Panjang: ", self.panjang * self.lebar)

    def keliling(self):
        print("Keliling Persegi Panjang: ", 2 * self.panjang + 2 * self.lebar)


class Segitiga(BangunDatar):
    def _init_(self, alas=0, tinggi=0):
        self.alas = alas
        self.tinggi = tinggi

    def luas(self):
        print("Luas Segitiga: ", 1/2 * self.alas * self.tinggi)


class Main():

    def menu():
        print("\n\nInheritance Program")
        print("1. Persegi")
        print("2. Lingkaran")
        print("3. Persegi Panjang")
        print("4. Segitiga")
        print("5. Keluar Program")

    menu()
    pilihan = int(input("\nPilih Menu:(Tulis Angka)"))

    while pilihan != 5:
        if pilihan == 1:
            persegi = Persegi()
            print("Menghitung Luas dan Keliling Persegi")
            persegi.sisi = int(input("Masukkan Sisi: "))
            persegi.luas()
            persegi.keliling()

        elif pilihan == 2:
            lingkaran = Lingkaran()
            print("Menghitung Luas dan Keliling Lingkaran")
            lingkaran.r = int(input("Masukkan Jari-jari: "))
            lingkaran.luas()
            lingkaran.keliling()

        elif pilihan == 3:
            persegiPanjang = PersegiPanjang()
            print("Menghitung Luas dan Keliling Persegi Panjang")
            persegiPanjang.panjang = int(input("Masukkan Panjang: "))
            persegiPanjang.lebar = int(input("Masukkan Lebar: "))
            persegiPanjang.luas()
            persegiPanjang.keliling()

        elif pilihan == 4:
            segitiga = Segitiga()
            print("Menghitung Luas Segitiga")
            segitiga.alas = int(input("Masukkan Alas: "))
            segitiga.tinggi = int(input("Masukkan Tinggi: "))
            segitiga.luas()

        else:
            print("Menu tidak tersedia")

        menu()
        pilihan = int(input("\nPilih Menu:(Tulis Angka) "))

    print("Terima Kasih telah menggunakan program!")

File: Inheritance.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Inheritance
{
    class Program
    {
        static void Main(string[] args)
        {
            Console.WriteLine("PROGRAM MENGHITUNG LUAS DAN KELILING BANGUN DATAR");

            Console.WriteLine();

            BangunDatar bangunDatar = new BangunDatar();

            Persegi p = new Persegi();
            p.sisi = 2;

            Lingkaran l = new Lingkaran();
            l.jarijari = 22;

            PersegiPanjang pp = new PersegiPanjang();
            pp.panjang = 8;
            pp.lebar = 4;

            Segitiga s = new Segitiga();
            s.alas = 12;
            s.tinggi = 8;

            bangunDatar.luas();
            bangunDatar.keliling();

            p.luas();
            p.keliling();

            l.luas();
            l.keliling();

            pp.luas();
            pp.keliling();

            s.luas();
            s.keliling();
        }

        public class BangunDatar
        {
            internal float luas()
            {
                Console.WriteLine("Menghitung luas bangun datar");
                return 0;
            }

            internal float keliling()
            {
                Console.WriteLine("Menghitung keliling bangun datar");
                return 0;
            }
        }

        public class Persegi : BangunDatar
        {
            private double s;
            public double sisi
            {
                get { return s; }
                set { s = value; }
            }
        }

        public class Lingkaran : BangunDatar
        {
            private double r;

            public double jarijari
            {
                get { return r; }
                set
                {
                    r = value;
                }
            }

        }

        public class PersegiPanjang : BangunDatar
        {
            private float p, l;

            public float panjang
            {
                get { return p; }
                set
                {
                    p = value;
                }
            }

            public float lebar
            {
                get { return l; }
                set
                {
                    l = value;
                }
            }
        }

        public class Segitiga : BangunDatar
        {
            private float a, t;

            public float alas
            {
                get { return a; }
                set
                {
                    a = value;
                }
            }

            public float tinggi
            {
                get { return t; }
                set
                {
                    t = value;
                }
            }
        }
    }
}